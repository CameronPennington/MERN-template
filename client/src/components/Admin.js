import React, { useEffect, useState } from "react";
import { connect, useDispatch } from "react-redux";
import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Checkbox from "@material-ui/core/Checkbox";
import Container from "@material-ui/core/Container";
import CssBaseline from "@material-ui/core/CssBaseline";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import Grid from "@material-ui/core/Grid";
import TextField from "@material-ui/core/TextField";

import Typography from "@material-ui/core/Typography";

import { makeStyles } from "@material-ui/core/styles";

import { validateUserData, validateEditUserData } from "../utils/validation";

import { setErrors } from "../actions/error";
import { addUser, getUsers, editUser, deleteUser } from "../actions/user";

const useStyles = makeStyles((theme) => ({
	adminHeader: {
		display: "flex",
		justifyContent: "flex-end",
	},
	form: {
		width: "100%", // Fix IE 11 issue.
		marginTop: theme.spacing(3),
	},
	paper: {
		marginTop: theme.spacing(8),
		display: "flex",
		flexDirection: "column",
		alignItems: "center",
	},
	submit: {
		margin: theme.spacing(3, 0, 2),
	},
}));

export const Admin = (props) => {
	const dispatch = useDispatch();
	const classes = useStyles();
	const [userList, setUserList] = useState([]);
	const [adminList, setAdminList] = useState([]);

	useEffect(() => {
		getUsers(props.user.properties.organization)
			.then(({ users }) => {
				// setOrgName(name);
				setUserList(users);
				const admins = users.filter((user) => user.admin);
				setAdminList(admins);
			})
			.catch((err) => {
				dispatch(setErrors(err));
			});
	}, []);

	const AddUserDialog = () => {
		const [open, setOpen] = useState(false);

		const [firstName, setFirstName] = useState("");
		const [firstNameHelper, setFirstNameHelper] = useState("");

		const [lastName, setLastName] = useState("");
		const [lastNameHelper, setLastNameHelper] = useState("");

		const [email, setEmail] = useState("");
		const [emailHelper, setEmailHelper] = useState("");

		const [password, setPassword] = useState("");
		const [passwordHelper, setPasswordHelper] = useState("");

		const [password2, setPassword2] = useState("");
		const [password2Helper, setPassword2Helper] = useState("");

		const [admin, setAdmin] = useState(false);

		const handleClickOpen = () => {
			setOpen(true);
		};

		const handleClose = () => {
			setOpen(false);
		};
		const onChange = (event) => {
			const { value, id } = event.target;
			switch (id) {
				case "firstName":
					setFirstNameHelper("");
					setFirstName(value);
					break;
				case "lastName":
					setLastNameHelper("");
					setLastName(value);
					break;
				case "email":
					setEmailHelper("");
					setEmail(value);
					break;
				case "password":
					setPasswordHelper("");
					setPassword(value);
					break;
				case "password2":
					setPassword2Helper("");
					setPassword2(value);
					break;
				default:
					break;
			}
		};

		const handleUserSubmit = (e) => {
			e.preventDefault();
			const orgId = props.user.properties.organization;
			const userData = {
				firstName,
				lastName,
				email,
				password,
				password2,
				organization: orgId,
			};
			const errors = validateUserData(userData);

			errors.firstName && setFirstNameHelper(errors.firstName);
			errors.lastName && setLastNameHelper(errors.lastName);
			errors.email && setEmailHelper(errors.email);
			errors.password && setPasswordHelper(errors.password);
			errors.password2 && setPassword2Helper(errors.password2);

			if (Object.keys(errors).length === 0) {
				addUser(orgId, userData)
					.then(({ users }) => {
						setUserList(users);
					})
					.catch((err) => {
						dispatch(setErrors(err));
					});
				setOpen(false);
			} else {
				return null;
			}
		};

		const handleCheck = (e) => {
			setAdmin(e.target.checked);
		};

		return (
			<div id="add-user-dialog">
				<Button
					variant="outlined"
					color="primary"
					onClick={handleClickOpen}
					id="add-user-button"
				>
					Add User
				</Button>
				<Dialog
					open={open}
					onClose={handleClose}
					aria-labelledby="form-dialog-title"
				>
					<DialogTitle id="form-dialog-title">Create New User</DialogTitle>
					<DialogContent>
						<form
							className={classes.form}
							noValidate
							onSubmit={handleUserSubmit}
							id="add-user-form"
						>
							<Grid container spacing={2}>
								<Grid item xs={12} sm={6}>
									<TextField
										autoComplete="fname"
										name="firstName"
										variant="outlined"
										required
										fullWidth
										id="firstName"
										label="First Name"
										autoFocus
										onChange={onChange}
										value={firstName}
										error={firstNameHelper.length > 0}
										helperText={firstNameHelper}
										inputProps={{ maxlength: 10 }}
									/>
								</Grid>
								<Grid item xs={12} sm={6}>
									<TextField
										variant="outlined"
										required
										fullWidth
										id="lastName"
										label="Last Name"
										name="lastName"
										autoComplete="lname"
										onChange={onChange}
										value={lastName}
										error={lastNameHelper.length > 0}
										helperText={lastNameHelper}
										inputProps={{ maxlength: 15 }}
									/>
								</Grid>
								<Grid item xs={12}>
									<TextField
										variant="outlined"
										required
										fullWidth
										id="email"
										label="Email Address"
										name="email"
										autoComplete="email"
										onChange={onChange}
										value={email}
										error={emailHelper.length > 0}
										helperText={emailHelper}
										inputProps={{ maxlength: 30 }}
									/>
								</Grid>

								<Grid item xs={12}>
									<TextField
										variant="outlined"
										required
										fullWidth
										name="password"
										label="Password"
										type="password"
										id="password"
										onChange={onChange}
										value={password}
										error={passwordHelper.length > 0}
										helperText={passwordHelper}
									/>
								</Grid>
								<Grid item xs={12}>
									<TextField
										variant="outlined"
										required
										fullWidth
										name="password2"
										label="Confirm Password"
										type="password"
										id="password2"
										onChange={onChange}
										value={password2}
										error={password2Helper.length > 0}
										helperText={password2Helper}
									/>
								</Grid>
								<Grid>
									<Checkbox
										disableRipple
										checked={admin}
										onChange={handleCheck}
										inputProps={{ "aria-label": "primary checkbox" }}
									/>
								</Grid>
							</Grid>
						</form>
					</DialogContent>
					<DialogActions>
						<Button onClick={handleClose} color="primary" id="cancel-add-task">
							Cancel
						</Button>
						<Button type="submit" color="primary" form="add-user-form">
							Create
						</Button>
					</DialogActions>
				</Dialog>
			</div>
		);
	};

	const UserCard = (props) => {
		//TODO: make it impossible for a user to remove their own admin status
		const EditDialog = (props) => {
			const [editOpen, setEditOpen] = useState(false);

			const [editFirstName, setEditFirstName] = useState(props.user.firstName);
			const [editFirstNameHelper, setEditFirstNameHelper] = useState("");

			const [editLastName, setEditLastName] = useState(props.user.lastName);
			const [editLastNameHelper, setEditLastNameHelper] = useState("");

			const [editEmail, setEditEmail] = useState(props.user.email);
			const [editEmailHelper, setEditEmailHelper] = useState("");

			const [editPassword, setEditPassword] = useState("");
			const [editPasswordHelper, setEditPasswordHelper] = useState("");

			const [admin, setAdmin] = useState(props.user.admin);

			const handleEditOpen = () => {
				setEditOpen(true);
			};
			const handleEditClose = () => {
				setEditOpen(false);
			};

			const onChange = (e) => {
				const { value, id } = e.target;
				switch (id) {
					case "firstName":
						setEditFirstNameHelper("");
						setEditFirstName(value);
						break;
					case "lastName":
						setEditLastNameHelper("");
						setEditLastName(value);
						break;
					case "email":
						setEditEmailHelper("");
						setEditEmail(value);
						break;
					case "password":
						setEditPasswordHelper("");
						setEditPassword(value);
						break;
					default:
						break;
				}
			};

			const handleCheck = (e) => {
				setAdmin(e.target.checked);
			};

			const handleEditSubmit = (e) => {
				e.preventDefault();
				const orgId = props.user.organization;
				const userId = props.user._id;

				const userData = {
					firstName: editFirstName,
					lastName: editLastName,
					email: editEmail,
					password: editPassword,
					admin: admin,
					organization: orgId,
				};
				const errors = validateEditUserData(userData);

				errors.firstName && setEditFirstNameHelper(errors.firstName);
				errors.lastName && setEditLastNameHelper(errors.lastName);
				errors.email && setEditEmailHelper(errors.email);
				errors.password && setEditPasswordHelper(errors.password);

				if (Object.keys(errors).length === 0) {
					editUser(orgId, userId, userData)
						.then(({ users }) => {
							setUserList(users);
						})
						.catch((err) => {
							dispatch(setErrors(err));
						});
					setEditOpen(false);
				} else {
					return null;
				}
			};

			return (
				<div id="add-user-dialog">
					<Button
						variant="outlined"
						color="primary"
						onClick={handleEditOpen}
						id="edit-user-button"
					>
						Edit User
					</Button>
					<Dialog
						open={editOpen}
						onClose={handleEditClose}
						aria-labelledby="form-dialog-title"
					>
						<DialogTitle id="form-dialog-title">Edit User</DialogTitle>
						<DialogContent>
							<form
								className={classes.form}
								noValidate
								onSubmit={handleEditSubmit}
								id="edit-user-form"
							>
								<Grid container spacing={2}>
									<Grid item xs={12} sm={6}>
										<TextField
											autoComplete="fname"
											name="firstName"
											variant="outlined"
											fullWidth
											id="firstName"
											label="First Name"
											autoFocus
											onChange={onChange}
											value={editFirstName}
											error={editFirstNameHelper.length > 0}
											helperText={editFirstNameHelper}
											inputProps={{ maxlength: 10 }}
										/>
									</Grid>
									<Grid item xs={12} sm={6}>
										<TextField
											variant="outlined"
											fullWidth
											id="lastName"
											label="Last Name"
											name="lastName"
											autoComplete="lname"
											onChange={onChange}
											value={editLastName}
											error={editLastNameHelper.length > 0}
											helperText={editLastNameHelper}
											inputProps={{ maxlength: 15 }}
										/>
									</Grid>
									<Grid item xs={12}>
										<TextField
											variant="outlined"
											fullWidth
											id="email"
											label="Email Address"
											name="email"
											autoComplete="email"
											onChange={onChange}
											value={editEmail}
											error={editEmailHelper.length > 0}
											helperText={editEmailHelper}
											inputProps={{ maxlength: 30 }}
										/>
									</Grid>

									<Grid item xs={12}>
										<TextField
											variant="outlined"
											fullWidth
											name="password"
											label="Password"
											type="password"
											id="password"
											onChange={onChange}
											value={editPassword}
											error={editPasswordHelper.length > 0}
											helperText={editPasswordHelper}
										/>
									</Grid>
									<Box>
										<Checkbox
											disableRipple
											disabled={
												props.loggedInUser.localeCompare(props.user._id) == 0
											}
											checked={admin}
											onChange={handleCheck}
											inputProps={{ "aria-label": "primary checkbox" }}
										/>
										Admin
									</Box>
								</Grid>
							</form>
						</DialogContent>
						<DialogActions>
							<Button
								onClick={handleEditClose}
								color="primary"
								id="cancel-edit-task"
							>
								Cancel
							</Button>
							<Button type="submit" color="primary" form="edit-user-form">
								Edit
							</Button>
						</DialogActions>
					</Dialog>
				</div>
			);
		};
		const DeleteDialog = (props) => {
			const [deleteOpen, setDeleteOpen] = useState(false);
			const handleDeleteOpen = () => {
				setDeleteOpen(true);
			};
			const handleDeleteClose = () => {
				setDeleteOpen(false);
			};
			const handleDeleteSubmit = (e) => {
				e.preventDefault();

				const orgId = props.user.organization;
				const userId = props.user._id;

				deleteUser(orgId, userId)
					.then(({ users }) => {
						setUserList(users);
					})
					.catch((err) => {
						dispatch(setErrors(err));
					});
				setDeleteOpen(false);
			};

			return (
				<>
					<Button variant="outlined" color="primary" onClick={handleDeleteOpen}>
						Delete User
					</Button>
					<Dialog open={deleteOpen} onClose={handleDeleteClose}>
						<DialogContent>
							<form
								noValidate
								onSubmit={handleDeleteSubmit}
								id="delete-user-form"
							>
								<DialogTitle>Are you sure?</DialogTitle>
								<DialogActions>
									{" "}
									<Button color="primary" onClick={handleDeleteClose}>
										Cancel
									</Button>
									<Button
										autoFocus
										color="primary"
										type="submit"
										form="delete-user-form"
									>
										Delete
									</Button>
								</DialogActions>
							</form>
						</DialogContent>
					</Dialog>
				</>
			);
		};

		return (
			<Card>
				<CardContent>
					<Typography>{`${props.user.firstName} ${props.user.lastName}`}</Typography>
				</CardContent>
				<CardActions>
					{" "}
					<EditDialog
						user={props.user}
						adminList={props.adminList}
						loggedInUser={props.loggedInUser}
					/>
					<DeleteDialog user={props.user} />
				</CardActions>
			</Card>
		);
	};
	return (
		<>
			<CssBaseline />
			<Container>
				<Box className={classes.adminHeader}>
					<AddUserDialog />
				</Box>
				{userList &&
					userList.length > 0 &&
					userList.map((user) => (
						<UserCard
							key={user._id}
							user={user}
							adminList={adminList}
							loggedInUser={props.user.properties.id}
						/>
					))}
			</Container>
		</>
	);
};

const mapStateToProps = (state) => ({
	user: state.user,
	errors: state.errors,
});

export default connect(mapStateToProps)(Admin);
