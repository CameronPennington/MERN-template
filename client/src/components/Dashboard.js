import React, { useState, useEffect } from "react";
import { connect, useDispatch } from "react-redux";
import { NavLink } from "react-router-dom";
import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import Card from "@material-ui/core/Card";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import Container from "@material-ui/core/Container";
import CssBaseline from "@material-ui/core/CssBaseline";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";
import TextField from "@material-ui/core/TextField";
import Typography from "@material-ui/core/Typography";
import useMediaQuery from "@material-ui/core/useMediaQuery";

import { makeStyles, useTheme } from "@material-ui/core/styles";
import ErrorSnackbar from "./ErrorSnackbar";

import { getOrg, addTask, deleteTask, editTask } from "../actions/organization";
import { setErrors } from "../actions/error";

const useStyles = makeStyles((theme) => ({
	organizationHeader: {
		display: "flex",
		marginBottom: "2rem",
	},
	orgTitleBox: {
		flexGrow: 1,
	},
	taskButtons: {
		display: "flex",
		justifyContent: "flex-end",
		width: "100%",
	},
	taskCard: {
		display: "inline-block",
		margin: "1rem 2rem 0 2rem",
		width: "40%",
		[theme.breakpoints.down("sm")]: {
			width: "100%",
		},
	},
	taskListBox: {
		display: "flex",
		flexWrap: "wrap",
	},
}));
//TODO: redirect to login on unauthenticated request
export const Dashboard = (props) => {
	const dispatch = useDispatch();
	const classes = useStyles();

	const [orgName, setOrgName] = useState("");
	const [taskList, setTaskList] = useState([]);

	const theme = useTheme();
	const matches = useMediaQuery(theme.breakpoints.down("sm"));

	useEffect(() => {
		getOrg(props.user.properties.organization)
			.then(({ orgName, tasks }) => {
				setOrgName(orgName);
				setTaskList(tasks);
			})
			.catch((err) => {
				dispatch(setErrors(err));
			});
	}, []);
	//TODO: should be able to pass tasklist in as a dependency
	const AddTaskDialog = () => {
		const [open, setOpen] = useState(false);
		const [taskName, setTaskName] = useState("");

		const handleClickOpen = () => {
			setOpen(true);
		};

		const handleClose = () => {
			setOpen(false);
		};
		const handleTaskSubmit = (e) => {
			e.preventDefault();
			const orgId = props.user.properties.organization;
			const taskData = { name: taskName };
			addTask(orgId, taskData)
				.then(({ tasks }) => {
					setTaskList(tasks);
				})
				.catch((err) => {
					dispatch(setErrors(err));
				});
			setOpen(false);
		};

		return (
			<div id="AddTaskDialog">
				<Button
					aria-label="Add Task"
					variant="outlined"
					color="primary"
					onClick={handleClickOpen}
					id="add-task"
				>
					<Typography variant="button" noWrap>
						Add Task
					</Typography>
				</Button>
				<Dialog
					open={open}
					onClose={handleClose}
					aria-labelledby="form-dialog-title"
				>
					<DialogTitle id="form-dialog-title">New Task</DialogTitle>
					<DialogContent>
						<form noValidate onSubmit={handleTaskSubmit} id="add-new-task">
							{/* TODO: reduce vertical spacing in ContentText */}
							<DialogContentText>
								To add a task, write the name of the task below
							</DialogContentText>
							<TextField
								autoFocus
								margin="dense"
								id="name"
								label="Name"
								fullWidth
								onChange={(e) => setTaskName(e.target.value)}
								value={taskName}
								inputProps={{ maxlength: 30 }}
							/>
						</form>
					</DialogContent>
					<DialogActions>
						<Button
							aria-label="Cancel Add Task"
							onClick={handleClose}
							color="primary"
							id="cancel-add-task"
						>
							Cancel
						</Button>
						<Button
							aria-label="Submit Add Task"
							type="submit"
							color="primary"
							form="add-new-task"
							disabled={taskName.length === 0}
						>
							Add
						</Button>
					</DialogActions>
				</Dialog>
			</div>
		);
	};
	const TaskCard = (props) => {
		const EditDialog = (props) => {
			const [editOpen, setEditOpen] = useState(false);
			const [editTaskName, setEditTaskName] = useState(props.name);
			const handleEditOpen = () => {
				setEditOpen(true);
			};
			const handleEditClose = () => {
				setEditOpen(false);
			};
			const handleEditSubmit = (e) => {
				e.preventDefault();
				const { orgId, taskId } = props;
				const taskData = { name: editTaskName };
				editTask(orgId, taskId, taskData)
					.then(({ tasks }) => {
						setTaskList(tasks);
					})
					.catch((err) => {
						dispatch(setErrors(err));
					});
			};

			return (
				<>
					<Button
						aria-label="Edit Task"
						variant="outlined"
						color="primary"
						onClick={handleEditOpen}
					>
						Edit
					</Button>
					<Dialog open={editOpen} onClose={handleEditClose}>
						<DialogTitle>Rename Task?</DialogTitle>
						<DialogContent>
							<form noValidate onSubmit={handleEditSubmit} id="edit-task-name">
								<TextField
									autoFocus
									margin="dense"
									id="edit-name"
									label="Name"
									fullWidth
									onChange={(e) => setEditTaskName(e.target.value)}
									value={editTaskName}
									inputProps={{ maxlength: 30 }}
								/>
								<DialogActions>
									{" "}
									<Button
										aria-label="Cancel Edit Task"
										color="primary"
										onClick={handleEditClose}
									>
										Cancel
									</Button>
									<Button
										aria-label="Submit Task Edit"
										color="primary"
										type="submit"
										form="edit-task-name"
										disabled={editTaskName.length === 0}
									>
										Edit
									</Button>
								</DialogActions>
							</form>
						</DialogContent>
					</Dialog>
				</>
			);
		};
		const DeleteDialog = (props) => {
			const [deleteOpen, setDeleteOpen] = useState(false);
			const handleDeleteOpen = () => {
				setDeleteOpen(true);
			};
			const handleDeleteClose = () => {
				setDeleteOpen(false);
			};
			const handleDeleteSubmit = (e) => {
				e.preventDefault();
				const { orgId, taskId } = props;
				deleteTask(orgId, taskId)
					.then(({ tasks }) => {
						setTaskList(tasks);
					})
					.catch((err) => {
						dispatch(setErrors(err));
					});
			};

			return (
				<>
					<Button
						aria-label="Delete Task"
						variant="outlined"
						color="primary"
						onClick={handleDeleteOpen}
					>
						Delete
					</Button>
					<Dialog open={deleteOpen} onClose={handleDeleteClose}>
						<DialogContent>
							<form noValidate onSubmit={handleDeleteSubmit} id="delete-task">
								<DialogTitle>Are you sure?</DialogTitle>
								<DialogActions>
									{" "}
									<Button
										aria-label="Cancel Delete Task"
										color="primary"
										onClick={handleDeleteClose}
									>
										Cancel
									</Button>
									<Button
										aria-label="Confirm Delete Task"
										autoFocus
										color="primary"
										type="submit"
										form="delete-task"
									>
										Delete
									</Button>
								</DialogActions>
							</form>
						</DialogContent>
					</Dialog>
				</>
			);
		};

		return (
			<>
				<Card className={classes.taskCard}>
					<CardContent>
						<Typography>{props.name}</Typography>
					</CardContent>

					<CardActions>
						{" "}
						<Box className={classes.taskButtons}>
							<Button
								variant="outlined"
								color="primary"
								component={NavLink}
								to={`/task/${props.taskId}`}
							>
								View
							</Button>{" "}
							<EditDialog
								taskId={props.taskId}
								orgId={props.orgId}
								name={props.name}
							/>
							<DeleteDialog taskId={props.taskId} orgId={props.orgId} />
						</Box>
					</CardActions>
				</Card>
			</>
		);
	};
	return (
		<>
			<CssBaseline />
			<Container>
				<Box className={classes.organizationHeader}>
					<Box className={classes.orgTitleBox}>
						<Typography variant={matches ? "h5" : "h4"}>{orgName}</Typography>
					</Box>
					<AddTaskDialog />
				</Box>
				<Box className={classes.taskListBox}>
					{taskList &&
						taskList.length > 0 &&
						taskList.map((task) => (
							<TaskCard
								key={task._id}
								name={task.name}
								taskId={task._id}
								orgId={props.user.properties.organization}
							/>
						))}
				</Box>
				<ErrorSnackbar errors={props.errors} />
			</Container>
		</>
	);
};

const mapStateToProps = (state) => ({
	user: state.user,
	errors: state.errors,
});

export default connect(mapStateToProps)(Dashboard);
