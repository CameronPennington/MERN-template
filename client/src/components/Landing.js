import React from "react";
import Box from "@material-ui/core/Box";
import CssBaseline from "@material-ui/core/CssBaseline";
import Typography from "@material-ui/core/Typography";

import { makeStyles } from "@material-ui/core/styles";

import Register from "./Register";

const useStyles = makeStyles((theme) => ({
	landingBox: {
		display: "flex",
		flexWrap: "wrap",
	},
	marketingBox: {
		display: "inline-block",
		flexGrow: 1,
	},
	registerBox: {
		display: "inline-block",
		flexGrow: 1,
	},
	textContainer: {
		display: "inline-block",
		maxWidth: "45rem",
	},
	textContainerWrapper: {
		alignItems: "center",
		display: "flex",
		height: "100%",
		justifyContent: "center",
		width: "100%",
	},
}));
//TODO: add margin to textContainer
export const Landing = () => {
	const classes = useStyles();

	const MarketingContent = () => {
		return (
			<>
				<Box className={classes.textContainerWrapper}>
					<Box className={classes.textContainer}>
						<Typography variant="h6">Why use Word?</Typography>
						<Typography variant="body1" paragraph>
							Why use a word processor to create training documents that will
							never be printed? No more adjusting margins and page breaks to get
							your content looking just right. With Docufy, you can create
							responsive web pages that look crisp and professional on any
							screen.
						</Typography>

						<Typography variant="h6">
							Knowledge management as a service
						</Typography>
						<Typography variant="body1" paragraph>
							Forget about maintaining your own intranet or a shared drive, now
							you can securely upload documents that are available anywhere.
						</Typography>

						<Typography variant="h6">Ease of use</Typography>
						<Typography variant="body1" paragraph>
							Simply type out the instructions for each task and upload any
							helpful images or screenshots. The result will be consistent and
							engaging content.
						</Typography>
					</Box>
				</Box>
			</>
		);
	};

	return (
		<>
			<CssBaseline />
			<Box className={classes.landingBox}>
				<Box className={classes.marketingBox}>
					<MarketingContent />
				</Box>
				<Box className={classes.registerBox}>
					<Register />
				</Box>
			</Box>
		</>
	);
};

export default Landing;
