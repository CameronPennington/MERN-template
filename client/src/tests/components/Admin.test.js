import React, { useEffect } from "react";
import {
	render,
	fireEvent,
	waitForElementToBeRemoved,
} from "@testing-library/react";
import { Admin } from "../../components/Admin";

describe("The admin component", () => {
	test("should render without crashing", () => {
		const props = {
			user: {
				isAuthenticated: true,
				properties: {
					organization: 1,
				},
			},
		};
		const { asFragment } = render(<Admin {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
