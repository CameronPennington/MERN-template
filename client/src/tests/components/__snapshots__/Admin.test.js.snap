// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The admin component should render without crashing 1`] = `<DocumentFragment />`;
